cranelift/entity/src/list.rs

```diff
@@ -865,4 +905,48 @@ mod tests {
         list.truncate(0, pool);
         assert!(list.is_empty());
     }
+
+    #[test]
+    fn copy_from() {
+        let pool = &mut ListPool::<Inst>::new();
+
+        let i1 = Inst::new(1);
+        let i2 = Inst::new(2);
+        let i3 = Inst::new(3);
+        let i4 = Inst::new(4);
+
+        let mut list = EntityList::from_slice(&[i1, i2, i3, i4], pool);
+        assert_eq!(list.as_slice(pool), &[i1, i2, i3, i4]);
+        let list2 = EntityList::from_slice(&[i4, i3, i2, i1], pool);
+        assert_eq!(list2.as_slice(pool), &[i4, i3, i2, i1]);
+        list.copy_from(&list2, 0..0, 0, pool);
+        assert_eq!(list.as_slice(pool), &[i1, i2, i3, i4]);
+        list.copy_from(&list2, 0..4, 0, pool);
+        assert_eq!(list.as_slice(pool), &[i4, i3, i2, i1, i1, i2, i3, i4]);
+    }
+
+    #[test]
+    #[should_panic]
+    fn copy_from_self() {
+        let pool = &mut ListPool::<Inst>::new();
+
+        let i1 = Inst::new(1);
+        let i2 = Inst::new(2);
+        let i3 = Inst::new(3);
+        let i4 = Inst::new(4);
+
+        let mut list = EntityList::from_slice(&[i4, i3, i2, i1, i1, i2, i3, i4], pool);
+        let list_again = list;
+        assert_eq!(list.as_slice(pool), &[i4, i3, i2, i1, i1, i2, i3, i4]);
+        list.copy_from(&list_again, 0..=1, 8, pool);
+        assert_eq!(
+            list.as_slice(pool),
+            &[i4, i3, i2, i1, i1, i2, i3, i4, i4, i3]
+        );
+        list.copy_from(&list_again, .., 7, pool);
+        assert_eq!(
+            list.as_slice(pool),
+            &[i4, i3, i2, i1, i1, i2, i4, i3, i2, i1, i1, i2, i3, i4, i4, i3, i3, i4, i4, i3]
+        )
```

Comment:
Which of these asserts is expected to panic?

Reply:
Neither; it would panic in the `copy_from`. I just copied them from the other test since they were examples that are now prohibited.

Reply:
Would you mind commenting the call where you expect the exception to happen, just so that it's super clear for anyone debugging the test?